src/ripple/shamap/SHAMap.h

```diff
@@ -182,10 +182,9 @@ class SHAMap
     
     bool getRootNode (Serializer & s, SHANodeFormat format) const;
     std::vector<uint256> getNeededHashes (int max, SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (uint256 const& hash, Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
-    SHAMapAddNode addRootNode (Blob const& rootNode, SHANodeFormat format,
-                               SHAMapSyncFilter * filter);
+    SHAMapAddNode addRootNode (boost::optional<uint256> const& hash,
```

Comment:
There are currently two `addRootNode` overloads with identical signatures except one is missing a `hash` argument.  The definitions of these two functions are nearly identical.  The one without the `hash` is called only from a test in SHAMapSync.test.cpp.
This P/R simply consolidates the two overloads into one, eliminating some repeated code.  The test now fills in the `hash` with `boost::none`.